modules/integration_azure-storage-account/variables-gen.tf

```diff
@@ -0,0 +1,310 @@
+# count detector
+
+variable "count_notifications" {
+  description = "Notification recipients list per severity overridden for count detector"
+  type        = map(list(string))
+  default     = {}
+}
+
+variable "count_aggregation_function" {
+  description = "Aggregation function and group by for count detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ".fill(None, duration='1d').count(by=['subscription_id'])"
+}
+
+variable "count_transformation_function" {
+  description = "Transformation function for count detector (i.e. \".mean(over='5m')\")"
+  type        = string
+  default     = ".min(over='1d')"
+}
+
+variable "count_tip" {
+  description = "Suggested first course of action or any note useful for incident handling"
+  type        = string
+  default     = ""
+}
+
+variable "count_runbook_url" {
+  description = "URL like SignalFx dashboard or wiki page which can help to troubleshoot the incident cause"
+  type        = string
+  default     = ""
+}
+
+variable "count_disabled" {
+  description = "Disable all alerting rules for count detector"
+  type        = bool
+  default     = null
+}
+
+variable "count_disabled_critical" {
+  description = "Disable critical alerting rule for count detector"
+  type        = bool
+  default     = null
+}
+
+variable "count_disabled_major" {
+  description = "Disable major alerting rule for count detector"
+  type        = bool
+  default     = null
+}
+
+variable "count_threshold_critical" {
+  description = "Critical threshold for count detector"
+  type        = number
+  default     = 245
+}
+
+variable "count_threshold_major" {
+  description = "Major threshold for count detector"
+  type        = number
+  default     = 240
+}
+
+# capacity detector
+
+variable "capacity_notifications" {
+  description = "Notification recipients list per severity overridden for capacity detector"
+  type        = map(list(string))
+  default     = {}
+}
+
+variable "capacity_aggregation_function" {
+  description = "Aggregation function and group by for capacity detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ".fill(None, duration='1d').sum(by=['azure_resource_name', 'azure_resource_group_name', 'azure_region'])"
```

Comment:
I understand why you do this. while it is not possible to use fill in current generator template you can trick it here.

I think a better approach would be to use a formula base on this simple metric signal and use fill inside this formula (and not the underlying metric signal).